# Tarea 2 - Portafolio de Inversión

**Integrantes:**  
María Fernanda Gómez Flores  
Carlos Hasan Hernández Rohan

**Periodo de análisis:** 01/01/2020 - 31/08/2026


In [1]:
#importar librerias
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import yfinance as yf

In [2]:
#Definir activos
tickers=['SPY', 'GLD', 'DBA', 'UUP']

In [3]:
#descargar datos
prices=yf.download(tickers,start='2020-01-01',end='2026-09-01')["Close"]
prices

/var/folders/x9/60hcc2ln62d3r3xyv90d3wl40000gn/T/ipykernel_84716/60511789.py:2: FutureWarning: YF.download() has changed argument auto_adjust default to True
  prices=yf.download(tickers,start='2020-01-01',end='2026-09-01')["Close"]
[*********************100%***********************]  4 of 4 completed


Ticker,DBA,GLD,SPY,UUP
Date,,,,
2020-01-02,14.560221,143.949997,296.125275,22.466349
2020-01-03,14.392557,145.860001,293.882935,22.492212
2020-01-06,14.427855,147.389999,295.004150,22.440483
2020-01-07,14.463151,147.970001,294.174683,22.518076
2020-01-08,14.392557,146.860001,295.742401,22.578423
...,...,...,...,...
2026-08-25,28.280001,428.070007,765.909973,27.940001
2026-08-26,28.590000,421.320007,766.080017,28.020000
2026-08-27,28.820000,422.600006,771.099976,28.020000


In [4]:
#rend diarios
rets=(prices/prices.shift()-1).dropna()
rets

Ticker,DBA,GLD,SPY,UUP
Date,,,,
2020-01-03,-0.011515,0.013269,-0.007572,0.001151
2020-01-06,0.002453,0.010490,0.003815,-0.002300
2020-01-07,0.002446,0.003935,-0.002812,0.003458
2020-01-08,-0.004881,-0.007502,0.005329,0.002680
2020-01-09,0.002453,-0.005652,0.006781,0.001909
...,...,...,...,...
2026-08-25,-0.000707,0.003234,0.003196,-0.000715
2026-08-26,0.010962,-0.015768,0.000222,0.002863
2026-08-27,0.008045,0.003038,0.006553,0.000000


## Rendimiento diario y anual del portafolio

In [5]:
#Rendimiento esperado de cada activo
mu=rets.mean()

In [6]:
#peso de los activos en el portafolio
w_port=np.array([0.25,0.25,0.25,0.25])

In [7]:
#Rendimiento del Portafolio
r_port=sum(w_port*mu)
r_port

0.0004860025444159994

In [8]:
#rend promedio
mu=rets.mean()*252
mu*100

Ticker
DBA    11.541007
GLD    17.424484
SPY    16.372996
UUP     3.650569
dtype: float64

In [9]:
#Rendimiento del Portafolio
r_port=sum(w_port*mu)
r_port

#es la suma ponderada del peso que tiene el activo en el portafolio por su rendimiento

0.12247264119283185

## Varianza diaria y anual del portafolio

In [10]:
#Riesgo portafolio
variance=rets.var()
variance

Ticker
DBA    0.000079
GLD    0.000135
SPY    0.000161
UUP    0.000021
dtype: float64

In [11]:
cov=rets.cov()
cov

Ticker,DBA,GLD,SPY,UUP
Ticker,,,,
DBA,0.000079,0.000015,0.000025,-0.000005
GLD,0.000015,0.000135,0.000022,-0.000023
SPY,0.000025,0.000022,0.000161,-0.000009
UUP,-0.000005,-0.000023,-0.000009,0.000021


In [12]:
daily_variance=w_port.T @ cov @ w_port

In [13]:
daily_variance

np.float64(2.7915276552673787e-05)

In [14]:
annual_variance=daily_variance*252
annual_variance*100

np.float64(0.7034649691273794)

## Volatilidad diaria y anual del portafolio

In [15]:
daily_volatility=np.sqrt(w_port.T @ cov @ w_port)

In [16]:
daily_volatility*100

np.float64(0.528349094374863)

In [17]:
volatility_annual=daily_volatility*np.sqrt(252)
volatility_annual*100

np.float64(8.387281854852498)

## Matriz de correlación

In [18]:
#Matriz de correlación
rets.corr()

Ticker,DBA,GLD,SPY,UUP
Ticker,,,,
DBA,1.000000,0.140796,0.222213,-0.110069
GLD,0.140796,1.000000,0.149491,-0.428662
SPY,0.222213,0.149491,1.000000,-0.153769
UUP,-0.110069,-0.428662,-0.153769,1.000000


## Conclusión sobre la matriz de correlación y el riesgo no sistemático de cada activo

### Matriz de correlación

La matriz muestra que los activos presentan correlaciones relativamente bajas entre sí. DBA, GLD y SPY tienen relaciones positivas pero débiles, mientras que UUP presenta correlaciones negativas con los otros tres activos.

La correlación más negativa se observa entre GLD y UUP, con un valor aproximado de -0.43, lo que indica que tienden a moverse en direcciones opuestas en cierta medida.

En general, la combinación de estos activos favorece la diversificación del portafolio, ya que no se encuentran altamente correlacionados entre sí.

### Riesgo no sistemático

Al comparar las varianzas individuales, SPY presenta el mayor nivel de riesgo entre los cuatro activos, seguido de GLD y DBA, mientras que UUP presenta la menor varianza.

Sin embargo, el riesgo del portafolio no depende solamente del riesgo individual de cada activo, sino también de la relación entre ellos. Como las correlaciones son en general bajas y algunas son negativas, la combinación de los cuatro activos permite reducir parte del riesgo no sistemático.

En especial, UUP ayuda a diversificar el portafolio porque presenta correlaciones negativas con DBA, GLD y SPY.


## Conclusión sobre las métricas de rendimiento, riesgo y diversificación del portafolio

El portafolio está formado por SPY, GLD, DBA y UUP, con una ponderación de 25% en cada activo.

La combinación presenta correlaciones relativamente bajas entre los activos y algunas relaciones negativas, especialmente entre GLD y UUP. Esto favorece la diversificación, ya que los activos no tienden a moverse todos en la misma dirección al mismo tiempo.

En cuanto al riesgo, la varianza diaria del portafolio obtenida es aproximadamente 0.0000279 y la volatilidad diaria es aproximadamente 0.53%. La volatilidad anual es aproximadamente 8.39%.

En general, el portafolio presenta una buena diversificación debido a la baja correlación entre los activos, lo que permite reducir parte del riesgo no sistemático. Después de ejecutar todas las celdas, se pueden incorporar aquí los valores finales del rendimiento diario, rendimiento anual y varianza anual mostrados en los resultados.
